tools/windows-node-installer/pkg/cloudprovider/factory.go

```diff
@@ -13,7 +16,33 @@ import (
 // this function requires specifying the credentialAccountID of the user's credential account.
 // The resourceTrackerDir is where the `windows-node-installer.json` file which contains IDs of created instance and
 // security group will be created.
-func CloudProviderFactory(kubeconfigPath, credentialPath, credentialAccountID, resourceTrackerDir string) (Cloud, error) {
+
+//struct to hold azure specific information.
+type svc struct {
+	resourceGroupName     string // name of the resource group
+	resourceGroupLocation string // name of the resource location
+
+	deploymentName string                 // name of the deployment
+	templateFile   map[string]interface{} // name of the templateFile (need from user)
+	parametersFile map[string]interface{} // name of the parametersFile (need from user)
+	authdata       map[string]interface{} // Azure auth data
+}
+
+func readJSON(path string) (map[string]interface{}, error) {
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		log.Fatal("failed to read file: %v", err)
+	}
+	contents := make(map[string]interface{})
+	json.Unmarshal(data, &contents)
+	return contents, nil
+}
+
+func CloudProviderFactory(
```

Comment:
Please don't make such changes. These inputs should remain on the same line. Moreover you inserted new code at the incorrect spot taking away the documentation existing for the function.